### ISSUE TITLE
New check: consider using any/all

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#5008]
### Current problem

Here is a function from Astroid:

```python
def is_from_decorator(node):
    """Return True if the given node is the child of a decorator"""
    for parent in node.node_ancestors():
        if isinstance(parent, Decorators):
            return True
    return False
```

What does it do? Evidently it loops over the node's ancestors and returns whether any of them are decorators. Well, why not just say that?

```python
def is_from_decorator(node):
    """Return True if the given node is the child of a decorator"""
    return any(isinstance(parent, Decorators)
               for parent in node.node_ancestors())
```

### Desired solution

The `any` and `all` functions are generally shorter and more declarative than explicit looping, so add a check to suggest using them when appropriate.

Implmenting this check so as to avoid too many false positives might be an interesting challenge!

**Comment by Pierre-Sassoulas** [comment]
I like this one because it's something I often look for during review and this will definitely automate a part of my review process.

**Comment by jTaylo99** [comment]
I'd like to give this a shot if that's alright.

**Comment by nickdrozd** [comment]
@jTaylo99 Please, go right ahead!

**Comment by jTaylo99** [comment]
Tried my luck for a few hours and made no headway, I think trying to build a new checker was a bit too big of a task for me to try and take on given I'm still working my head around the source code and astroid. Good luck to anyone else who wants to implement this as I think it'll be super helpful!
